src/Symfony/Component/Process/Process.php

```diff
@@ -278,7 +278,10 @@ public function start(callable $callback = null, array $env = array())
 
         $options = array('suppress_errors' => true);
 
-        if ('\\' === DIRECTORY_SEPARATOR) {
+        $windows = '\\' === DIRECTORY_SEPARATOR;
+        $commandline = $this->replacePlaceholder($commandline, $windows ? array('!', '!') : array('$', ''));
```

Comment:
the "$toReplace" should be moved inside the method, there is no reason to leak this outside of the method I think